tests are currently broken

It seems that tests fails on HEAD since 0ef00ee. Here's the travis build log:

https://travis-ci.org/wummel/linkchecker/builds/104276729

This affects all pull requests which also fail to pass tests.

this was previously reported in https://github.com/wummel/linkchecker/issues/681

For some reason Travis excludes some of the tests.  If I run the full test suite (`make test` without the `TESTS="<explicit subset specified in .travis.yml>"`), I see four test failures:

```
====================================================================== FAILURES ======================================================================
__________________________________________________________ TestConsole.test_internal_error ___________________________________________________________

self = <tests.test_console.TestConsole testMethod=test_internal_error>

    def test_internal_error (self):
>       linkcheck.director.console.internal_error()

tests/test_console.py:29: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
linkcheck/director/console.py:99: in internal_error
    print_app_info(out=out)
linkcheck/director/console.py:152: in print_app_info
    print(configuration.get_modules_info(), file=out)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def get_modules_info():
        """Return unicode string with detected module info."""
        module_infos = []
        for (mod, name, version_attr) in Modules:
            if not fileutil.has_module(mod):
                continue
>           if hasattr(mod, version_attr):
E           TypeError: hasattr(): attribute name must be string

linkcheck/configuration/__init__.py:84: TypeError
---------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------


********** Oops, I did it again. *************

You have found an internal error in LinkChecker. Please write a bug report
at https://github.com/wummel/linkchecker/issues
and include the following information:
- the URL or file you are testing
- the system information below

When using the commandline client:
- your commandline arguments and any custom configuration files.
- the output of a debug run with option "-Dall"

Not disclosing some of the information above due to privacy reasons is ok.
I will try to help you nonetheless, but you have to give me something
I can work with ;) .

Traceback (most recent call last):
None
System info:
LinkChecker 9.4
Released on: xx.xx.xxxx
Python 2.7.12 (default, Jul  1 2016, 15:12:24) 
[GCC 5.4.0 20160609] on linux2
PYTHONHASHSEED = '2364307325'
__________________________________________________________ TestFile.test_directory_listing ___________________________________________________________

self = <tests.checker.test_file.TestFile testMethod=test_directory_listing>

    def test_directory_listing (self):
        # unpack non-unicode filename which cannot be stored
        # in the SF subversion repository
        if os.name != 'posix' or sys.platform != 'linux2':
            return
        dirname = get_file("dir")
        if not os.path.isdir(dirname):
            unzip(dirname + ".zip", os.path.dirname(dirname))
>       self.file_test("dir")

tests/checker/test_file.py:91: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/checker/__init__.py:199: in file_test
    self.fail_unicode(msg)
tests/checker/__init__.py:205: in fail_unicode
    self.fail(msg)
E   AssertionError: tests/checker/data/dir
E   --- 
E   
E   +++ 
E   
E   @@ -3,8 +3,8 @@
E   
E    real url file:///home/mg/src/linkchecker/tests/checker/data/dir/
E    name tests/checker/data/dir
E    valid
E   -url %C3%AD%C2%BB%C2%AD%C2%AF%C2%BF.dat
E   -cache key file:///home/mg/src/linkchecker/tests/checker/data/dir/%C3%AD%C2%BB%C2%AD%C2%AF%C2%BF.dat
E   -real url file:///home/mg/src/linkchecker/tests/checker/data/dir/%C3%AD%C2%BB%C2%AD%C2%AF%C2%BF.dat
E   -name ??????????.dat
E   +url %ED%BB%AD%AF%BF.dat
E   +cache key file:///home/mg/src/linkchecker/tests/checker/data/dir/%ED%BB%AD%AF%BF.dat
E   +real url file:///home/mg/src/linkchecker/tests/checker/data/dir/%ED%BB%AD%AF%BF.dat
E   +name ?????.dat
E    valid
_________________________________________________________________ TestHttp.test_html _________________________________________________________________

self = <tests.checker.test_http.TestHttp testMethod=test_html>

    def test_html (self):
        confargs = dict(recursionlevel=1)
>       self.file_test("http.html", confargs=confargs)

tests/checker/test_http.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/checker/__init__.py:199: in file_test
    self.fail_unicode(msg)
tests/checker/__init__.py:205: in fail_unicode
    self.fail(msg)
E   AssertionError: http://localhost:33267/tests/checker/data/http.html
E   --- 
E   
E   +++ 
E   
E   @@ -39,13 +39,13 @@
E   
E    real url http://localhost:33267/?q=%C3%BC
E    name html entities
E    valid
E   +url file.html
E   +cache key http://localhost:33267/tests/checker/data/file.html
E   +real url http://localhost:33267/tests/checker/data/file.html
E   +valid
E    url file.css
E    cache key http://localhost:33267/tests/checker/data/file.css
E    real url http://localhost:33267/tests/checker/data/file.css
E   -valid
E   -url file.html
E   -cache key http://localhost:33267/tests/checker/data/file.html
E   -real url http://localhost:33267/tests/checker/data/file.html
E    valid
E    url file.asc
E    cache key http://localhost:33267/tests/checker/data/file.asc
________________________________________________________________ TestHttps.test_https ________________________________________________________________

args = (<tests.checker.test_https.TestHttps testMethod=test_https>,), kwargs = {}

    def newfunc (*args, **kwargs):
        if not testfunc():
            pytest.skip("%s is not available" % name)
>       return func(*args, **kwargs)

tests/__init__.py:90: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/checker/test_https.py:45: in test_https
    self.direct(url, resultlines, recursionlevel=0, confargs=confargs)
tests/checker/__init__.py:228: in direct
    self.fail_unicode(unicode(os.linesep).join(l))
tests/checker/__init__.py:205: in fail_unicode
    self.fail(msg)
E   AssertionError: Differences found testing https://www.amazon.com/
E   @@ -1,5 +1,4 @@
E    url https://www.amazon.com/
E    cache key https://www.amazon.com/
E    real url https://www.amazon.com/
E   -info Access denied by robots.txt, checked only syntax.
E    valid
================================================= 4 failed, 205 passed, 15 skipped in 53.98 seconds ==================================================
```

--- a/linkcheck/configuration/__init__.py
+++ b/linkcheck/configuration/__init__.py
@@ -81,7 +81,7 @@ def get_modules_info():
     for (mod, name, version_attr) in Modules:
         if not fileutil.has_module(mod):
             continue
-        if hasattr(mod, version_attr):
+        if version_attr and hasattr(mod, version_attr):
             attr = getattr(mod, version_attr)
             version = attr() if callable(attr) else attr
             module_infos.append("%s %s" % (name, version))

